### PR TITLE
Added unregister change handler to useEffect

### LINKED ISSUE #4344: Bug: New sidebar categories do not display tooltip on DND, and deleted categories persist on sidebar
## Steps to reproduce the behavior

1. Login as a new user
2. Create a new category on the sidebar
3. See error - no tooltip displays to indicate DND
4. Delete the category
5. See error - category does not disappear and now displays tooltip for DND
6. Seems to only happen for newly logged in users. After refreshing a few times, the issue no longer occurs

## Expected behavior

New categories display the `New` label and a tooltip for DND. Deleted categories are removed from the sidebar.

## Screenshots (optional)

https://user-images.githubusercontent.com/93531870/207675675-9e544a9d-9830-43b4-bb75-bcfdf2f8e699.mov

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: v7.7
 - Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary

The issue was being caused by duplicate change handlers being registered for handling category change.
Since the change the handler marks any existing category as "not new" if a change occurs in it, the second time the duplicate handler was called, it saw the category as already existing and marked it as not new.
Returning a cleanup function in `useEffect` to unregister the WebSocket change handler resolves the issue.

#### Ticket Link

Fixes #4344

### CODE REVIEW

**Review by sbishel — APPROVED** [review]

**Comment by harshilsharma63** [comment]
@ogi-m can you please test this? I'll merge it once tested and approved by you.

**Review by ogi-m — APPROVED** [review]
Tested and passed
- "new" label and dnd tooltip are displayed as expected when creating a new category
- deleting a category works as expected

**Pull request merged**
